File: api/schemas/account.py
import re
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime
from models.account import MembershipRole


def generate_slug(name: str) -> str:
    slug = name.lower()
    slug = re.sub(r"[^a-z0-9\s-]", "", slug)
    slug = re.sub(r"[\s]+", "-", slug)
    slug = re.sub(r"-+", "-", slug)
    return slug.strip("-")[:100]


class AccountCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=255)
    slug: Optional[str] = Field(None, min_length=1, max_length=100)


class AccountResponse(BaseModel):
    id: int
    name: str
    slug: str
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class MembershipResponse(BaseModel):
    id: int
    user_id: int
    account_id: int
    role: MembershipRole
    created_at: datetime
    account: AccountResponse

    class Config:
        from_attributes = True


class AddMemberRequest(BaseModel):
    email: str
    role: MembershipRole = MembershipRole.MEMBER


class SwitchAccountRequest(BaseModel):
    account_id: int


class SlugValidationResponse(BaseModel):
    slug: str
    available: bool

File: admin/src/types/account.ts
export interface Account {
  id: number
  name: string
  slug: string
  created_at: string
  updated_at: string
}

export type MembershipRole = 'admin' | 'member'

export interface AccountMembership {
  id: number
  user_id: number
  account_id: number
  role: MembershipRole
  created_at: string
  account: Account
}

export interface CurrentUser {
  id: number
  email: string
  username: string
  first_name: string | null
  last_name: string | null
  role: string
  is_active: boolean
  is_superadmin: boolean
  last_active_account_id: number | null
  lead_score: number
  user_metadata: Record<string, unknown>
  created_at: string
  updated_at: string
}

export interface SlugValidation {
  slug: string
  available: boolean
}
